lib/libc/common/source/stdlib/malloc.c

```diff
@@ -260,7 +260,7 @@ void free(void *ptr)
 	malloc_unlock();
 }
 
-SYS_INIT(malloc_prepare, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
+SYS_INIT(malloc_prepare, POST_KERNEL, 0);
```

Comment:
Not blocking but wouldn't be better having a specific symbol for the C library ?
```
config KERNEL_INIT_PRIORITY_LIBC
	int "LIBC initialization priority"
	default 30
	help
	  LIBC uses this priority for initialization. This
	  priority needs to be higher than minimal default initialization
	  priority.
```

Reply:
Hi @ceolin , changed as your comment, pls review again